master/internal/rm/agentrm/agent.go

```diff
@@ -873,6 +873,12 @@ func (a *agent) socketDisconnected() {
 	timer := time.AfterFunc(a.agentReconnectWait, a.HandleReconnectTimeout)
 	a.reconnectTimers = append(a.reconnectTimers, timer)
 
+	defer a.notifyListeners()
```

Comment:
stray change?

Reply:
there were actually a bunch of null pointer exceptions showing up in testing, they seemed unrelated to this feature in general but maybe something that needs to be looked into.